pandas/core/series.py

```diff
@@ -1074,15 +1075,30 @@ def tolist(self):
         """ Convert Series to a nested list """
         return list(self.asobject)
 
-    def to_dict(self):
+    def to_dict(self, into=dict):
         """
-        Convert Series to {label -> value} dict
+        Convert Series to {label -> value} dict or dict-like object
+        Parameters
+        ----------
+        into : class, default dict
+            The collections.Mapping subclass to use as the return
+            object.
+            .. versionadded:: 0.20.0
 
         Returns
         -------
-        value_dict : dict
-        """
-        return dict(compat.iteritems(self))
+        value_dict : collections.Mapping
+            If ``into`` is collections.defaultdict, the return
+            value's default_factory will be None.
+        """
+        # GH16122
+        if issubclass(into, collections.Mapping):
+            if into == collections.defaultdict:
+                return into(None, compat.iteritems(self))
+            else:
```

Comment:
make this and the code in frame into a helper function to avoid duplication
u can put this is core.common